apps/store/src/utils/useResponsiveVariant.ts

```diff
@@ -0,0 +1,14 @@
+import { useEffect, useState } from 'react'
+import { type Level } from 'ui'
+import { useBreakpoint } from './useBreakpoint/useBreakpoint'
+
+// Lazy responsive rendering, we don't want hydration errors
+// and we don't want user-agent dependency for server side (mainly because it's not compatible with SSG)
+export const useResponsiveVariant = (breakpoint: Level) => {
+  const [variant, setVariant] = useState<'desktop' | 'mobile' | 'initial'>('initial')
```

Comment:
nit:
```suggestion
  const [variant, setVariant] = useState<'desktop' | 'mobile' | 'unkown'>('unkown')
```
What do you think about this change?

Reply:
I think 'unknown' is a bit misleading, we always know variant after initial client rendering.  Not a fan of 'initial' either, would be happy to improve it

Reply:
I see. I was thinking on that 'initial' value as something we only use on the server as the placeholder for the period we can't know which variant we're dealing with. That's why 'unknow'.